tests/end-to-end/simple/app.py

```diff
@@ -0,0 +1,131 @@
+    super().__init__(scope, construct_id, **kwargs)
+
+    # Mappings
+    booleans = {
+      'True': {
+        'true': True,
+      },
+      'False': {
+        'false': False,
+      },
+    }
+    lists = {
+      'Candidates': {
+        'Empty': [],
+        'Singleton': ['One'],
+        'Pair': ['One','Two'],
+      },
+    }
+    numbers = {
+      'Prime': {
+        'Eleven': 11,
+        'Thirteen': 13,
+        'Seventeen': 17,
+      },
+    }
+    strings = {
+      'Foos': {
+        'Foo1': 'Foo1',
+        'Foo2': 'Foo2',
+      },
+      'Bars': {
+        'Bar': 'Bar',
+      },
+    }
+    table = {
+      'Values': {
+        'Boolean': True,
+        'Float': 3.14,
+        'List': ['1','2','3'],
+        'Number': 42,
+        'String': 'Baz',
+      },
+    }
+
+    # Conditions
+    is_us = cdk.Fn.select(0, self.region.split('-')) == 'us'
+    is_us_east1 = self.region == 'us-east-1'
+    is_large_region = is_us_east1
+
+    # Resources
+    queue = sqs.CfnQueue(self, 'Queue',
+          delay_seconds = 42.1337,
+          fifo_queue = False,
+          kms_master_key_id = cdk.Fn.importValue('Shared.KmsKeyArn'),
+          queue_name = [
+            self.stackName,
+            strings['Bars']['Bar'],
+            cdk.Fn.select(1, cdk.Fn.getAzs(self.region)),
+          ].join('-'),
+          redrive_policy = None,
+          visibility_timeout = 120,
+        )
+
+    if (queue is None): raise Exception("A combination of conditions caused 'queue' to be None. Fixit.")
+    bucket = s3.CfnBucket(self, 'Bucket',
+          access_control = 'private',
+          bucket_name = '{props.bucketNamePrefix}-{self.stackName}-bucket',
+          logging_configuration = {
+            'destinationBucketName': props.logDestinationBucketName,
+          },
+          website_configuration = {
+            'indexDocument': 'index.html',
+            'errorDocument': 'error.html',
+            'redirectAllRequestsTo': {
+              'hostName': 'example.com',
+              'protocol': 'https',
+            },
+          },
+          tags = [
+            {
+              'key': 'FancyTag',
+              'value': cdk.Fn.base64(table['Values']['String']) if is_us_east1 else _buffer.from('8CiMvAo=', 'base64').toString('binary'),
+            },
+          ],
+        ) if is_us_east1 else None
+    if (bucket is not None):
+      bucket.cfnOptions.metadata = {
+        CostCenter: 1337,
+      }
+      bucket.cfnOptions.deletionPolicy = cdk.CfnDeletionPolicy.RETAIN
+      bucket.addDependency(queue)
+
+    # Outputs
+    self.bucket_arn = bucket.attrarn if is_us_east1 else None
+    if (is_us_east1):
+      cdk.CfnOutput(self, 'BucketArn', 
+        description = 'The ARN of the bucket in this template!',
+        export_name = 'ExportName',
+        value = self.bucket_arn,
+      )
+
+    self.queue_arn = queue.ref
+    self.is_large = True if is_large_region else False
```

Comment:
Sean and I synced just now on this and these should also be `CfnOutput` the difference here is that we don't want to include `export_name` on these two. We also don't want to assign these as class fields because CDK logic will then turn them into exports.